### PR TITLE
Adding Environment Pararmeter to enable script to run against AzureUSGovernment

### PR DESCRIPTION
**Pull request by mrpullen**
Updating the script to allow a user to provide an environment argument so that AzureUSGovernment and other supported Environments can be supported.

### CODE REVIEW

File: AzureMonitor/Scripts/Create-AzDiagPolicy.ps1 [hunk 1]
```diff
@@ -437,7 +450,7 @@ function Get-ResourceType (
                 catch
                 {
                     # Uncomment below to see actual error.  Certain resources are not ResourceTypes that can support Logs and Metrics so the host error is being muted
-                    #write-host $Error[0] -ForegroundColor Red
+                    # write-host $Error[0] -ForegroundColor Red
```
Comment (JimGBritt) [inline AzureMonitor/Scripts/Create-AzDiagPolicy.ps1]: Can we remove the space? That way someone removes the comment and proper indenting occurs. No need to add the space.

File: AzureMonitor/Scripts/Create-AzDiagPolicy.ps1 [hunk 2]
```diff
@@ -1928,8 +1941,9 @@ if($ManagementGroup)
     $SubScriptionsToProcess =@()
     if($ManagementGroupID)
     {
+        $azEnvironment = Get-AzEnvironment -Name $Environment
         $GetBody = BuildBody -method "GET"
-        $MGSubsDetailsURI = "https://management.azure.com/providers/microsoft.management/managementGroups/$($ManagementGroupID)/descendants?api-version=2018-03-01-preview"
+        $MGSubsDetailsURI = "https://$($azEnvironment.ResourceManagerUrl)/providers/microsoft.management/managementGroups/$($ManagementGroupID)/descendants?api-version=2018-03-01-preview"
```
Comment (JimGBritt) [inline AzureMonitor/Scripts/Create-AzDiagPolicy.ps1:1946]: Please update to

$MGSubsDetailsURI = "$($azEnvironment.ResourceManagerUrl)providers/microsoft.management/managementGroups/$($ManagementGroupID)/descendants?api-version=2018-03-01-preview"

File: AzureMonitor/Scripts/Trigger-PolicyInitiativeRemediation.ps1 [hunk 3]
```diff
@@ -290,8 +297,9 @@ if($ManagementGroup)
     $SubScriptionsToProcess =@()
     if($ManagementGroupID)
     {
+        $azEnvironment = Get-AzEnvironment -Name $Environment
         $GetBody = BuildBody -method "GET"
-        $MGSubsDetailsURI = "https://management.azure.com/providers/microsoft.management/managementGroups/$($ManagementGroupID)/descendants?api-version=2018-03-01-preview"
```
Comment (JimGBritt) [inline AzureMonitor/Scripts/Trigger-PolicyInitiativeRemediation.ps1:294]: This currently doesn't appear to be working properly. See comment in Create-AzDiagPolicy.PS1  Needs to be

$MGSubsDetailsURI = "$($azEnvironment.ResourceManagerUrl)providers/microsoft.management/managementGroups/$($ManagementGroupID)/descendants?api-version=2018-03-01-preview"

**Review by JimGBritt — COMMENTED** [review]
Please take a look.  Just a couple of changes to fix some logic / format

File: AzureMonitor/Scripts/Create-AzDiagPolicy.ps1 (hunk 1 above)
Comment (mrpullen) [reply AzureMonitor/Scripts/Create-AzDiagPolicy.ps1]: Done in 
3d3d363

File: AzureMonitor/Scripts/Create-AzDiagPolicy.ps1 (hunk 2 above)
Comment (mrpullen) [reply AzureMonitor/Scripts/Create-AzDiagPolicy.ps1:1946]: Done in 

ebe972d

File: AzureMonitor/Scripts/Trigger-PolicyInitiativeRemediation.ps1 (hunk 3 above)
Comment (mrpullen) [reply AzureMonitor/Scripts/Trigger-PolicyInitiativeRemediation.ps1:294]: Done in 
ebe972d

**Review by JimGBritt — APPROVED** [review]
Looks good Mike thank you.

**Pull request merged**